.github/workflows/ci.yml

```diff
@@ -87,10 +100,23 @@ jobs:
       # Action for this, as of Jan 2021 it does not support Python coverage
       # files - only lcov files.  Therefore, we use coveralls-python, the
       # coveralls.io-supplied Python reporter, for this.
-      - name: "Report Coverage to Coveralls"
+      #
+      # If we `pip install coveralls` (with Python 2.7), we will get
+      # coveralls-python 1.11.1.  If we `pip3 install coveralls` (with
+      # Python 3), we will get coveralls-python 3.0.1.
+      #
+      # The two versions are somewhat incompatible, so mixing these
+      # two different versions when reporting coverage to coveralls.io
+      # will lead to grief: they get job IDs in different fashions, so
+      # in the same workflow, this finalizing step will be only able
+      # to set half of the jobs as done.  Therefore we must try to use
+      # the newer coveralls-python that is available to Python 3
+      # throughout this workflow.
+      - name: Report coverage to coveralls.io
```

Comment:
Whenever writing yaml, I quote strings.  This avoids the surprising case of `foo: true` and "foo: on" and so on.

Reply:
Ouch.  None of the names in this file are quoted.  GitHub's [docs](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions) for workflow syntax does not quote names.  I merely followed that convention.

Should I fix them all in this PR?

Reply:
I wouldn't mix that global change with these changes.  It might be worth a separate PR.  Or not.  Since GitLab and all.